.storybook/decorators/context.js

```diff
@@ -8,81 +8,117 @@ import { html } from "lit";
 export const withContextWrapper = makeDecorator({
 	name: "withContextWrapper",
 	parameterName: "context",
-	wrapper: (StoryFn, context) => {
-		const { args = {}, argTypes = {}, viewMode, id, loaded = {} } = context;
-		const { tokens = {} } = loaded;
-
-		const getDefaultValue = (type) => {
-			if (!type) return null;
-			if (type.defaultValue) return type.defaultValue;
-			return type.options ? type.options[0] : null;
+	wrapper: (StoryFn, data) => {
+		const {
+			args: {
+				rootClass,
+				staticColor,
+				...args
+			} = {},
+			globals: {
+				color = "light",
+				context = "spectrum1",
+				scale = "medium",
+			} = {},
+			viewMode,
+			id,
+			loaded: {
+				tokens = {},
+			} = {}
+		} = data;
+
+		const [, updateGlobals] = useGlobals();
+		const [, updateArgs] = useArgs();
+
+		const staticColorSettings = {
+			"black": {
+				background: "var(--spectrum-docs-static-black-background-color)",
+				color: "light"
+			},
+			"white": {
+				background: "var(--spectrum-docs-static-white-background-color)",
+				color: "dark"
+			},
 		};
 
-		// This property informs which context stylesheets to source
-		//    but does not source a stylesheet for itself
-		/** @type boolean */
-		const isExpress = args.express
-			? args.express
-			: getDefaultValue(argTypes.express);
-		/** @type string */
-		const color = args.color ? args.color : getDefaultValue(argTypes.color) ?? "light";
-		/** @type string */
-		const scale = args.scale ? args.scale : getDefaultValue(argTypes.scale) ?? "medium";
+		/**
+		 * @deprecated allow temporary fallback support for values defined in the args
+		 * */
+		if (args.color && args.color !== color) {
+			updateGlobals({ color: args.color });
+			// prevents unnecessary re-renders
+			updateArgs({ color: undefined });
+		}
+
+		if (args.express && context !== "express") {
+			updateGlobals({ context: "express" });
+			// prevents unnecessary re-renders
+			updateArgs({ express: undefined });
+		}
+
+		if (args.scale && args.scale !== scale) {
+			updateGlobals({ scale: args.scale });
+			// prevents unnecessary re-renders
+			updateArgs({ scale: undefined });
+		}
+
+		window.__color = color;
+		window.__context = context;
+		window.__scale = scale;
```

Comment:
Could you explain why these globals are being stored in `window` now, and some components (e.g. combobox, datepicker, slider) that were previously `useGlobals()` are now accessing the window object?